OfficeMicroService/Application/Services/OfficeServices.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AutoMapper;
+using MongoDB.Driver;
+using OfficeMicroService.Application.Exceptions;
+using OfficeMicroService.Application.Services.DTO;
+using OfficeMicroService.Data.Enum;
+using OfficeMicroService.Data.Models;
+using OfficeMicroService.Data.Repository;
+
+namespace OfficeMicroService.Application.Services
+{
+    public class OfficeServices : IOfficeServices
+    {
+        private readonly IOfficeRepository _repository;
+        private readonly IMapper _mapper;
+
+        public OfficeServices(IOfficeRepository repository, IMapper mapper)
+        {
+            _repository = repository;
+            _mapper = mapper;
+        }
+
+        public async Task<List<OfficeDTO>> GetAllAsync()
+        {
+            var offices = (await _repository.FindByCondition(office => true));
+            return _mapper.Map<List<OfficeDTO>>(offices);
+        }
+        public async Task<OfficeDTO> GetAsync(string id)
+        {
+            var office = (await _repository.FindByCondition(office => office.Id == id)).FirstOrDefault();
+            return _mapper.Map<OfficeDTO>(office);
+        }
+        public async Task<OfficeDTO> CreateAsync(OfficeForChangeDTO model)
+        {
+            var mapModel = _mapper.Map<Office>(model);
+            if (mapModel == null)
```

Comment:
The only situation when it's null is when model is null. So you need to remove this map & check and add check if model is null (before mapping) . Please change it in other places also